GitUI/UserControls/FileStatusList.cs

```diff
@@ -620,91 +621,129 @@ public void SelectStoredNextIndex(int defaultIndex = -1)
+                    var baseRevGuid = Module.GetMergeBase(GetRevisionOrHead(firstRev, head),
+                        GetRevisionOrHead(selectedRev, head));
+
+                    // Add if separate branches (note that artificial commits both have HEAD as BASE)
+                    if (baseRevGuid != null
+                        && baseRevGuid != GetRevisionOrHead(firstRev, head)
+                        && baseRevGuid != GetRevisionOrHead(selectedRev, head))
                     {
-                        // Get base commit, add as parent if unique
-                        Lazy<ObjectId> head = getRevision != null
-                            ? new Lazy<ObjectId>(() => getRevision(ObjectId.IndexId).FirstParentGuid)
-                            : new Lazy<ObjectId>(() => Module.RevParse("HEAD"));
-                        var revA = parentRevs[0].ObjectId;
-                        var revB = Revision.ObjectId;
-                        baseRevGuid = Module.GetMergeBase(GetRevisionOrHead(revA, head),
-                            GetRevisionOrHead(revB, head));
-                        if (baseRevGuid != null
-                            && baseRevGuid != revA
-                            && baseRevGuid != revB)
-                        {
-                            Array.Resize(ref parentRevs, 2);
-                            parentRevs[1] = new GitRevision(baseRevGuid);
-                        }
+                        // Present common files in BASE->B, BASE->A separately
+                        // For the following diff:  A->B a,c,d; BASE->B a,b,c; BASE->A a,b,d
+                        // (the file a has unique changes, b has the same change and c,d is changed in one of the branches)
+                        // The following groups will be shown: A->B a,c,d; BASE->B a,c; BASE->A a,d; Common BASE b
+                        var allBaseToB = Module.GetDiffFilesWithSubmodulesStatus(baseRevGuid, selectedRev.ObjectId,
+                            selectedRev.FirstParentGuid);
+                        var allBaseToA = Module.GetDiffFilesWithSubmodulesStatus(baseRevGuid, firstRev.ObjectId,
+                            firstRev.FirstParentGuid);
+
+                        var gitItemStatusNameComparer = new GitItemStatusNameComparer();
+                        var commonBaseToAandB = allBaseToB.Intersect(allBaseToA, gitItemStatusNameComparer)
+                            .Except(allAToB, gitItemStatusNameComparer).ToList();
+                        var uniqueBaseToB = allBaseToB.Except(commonBaseToAandB, gitItemStatusNameComparer).ToList();
+                        var uniqueBaseToA = allBaseToA.Except(commonBaseToAandB, gitItemStatusNameComparer).ToList();
+
+                        var revBase = new GitRevision(baseRevGuid);
+                        tuples.Add((revBase,
+                            _diffBaseToB.Text + GetDescriptionForRevision(selectedRev.ObjectId), uniqueBaseToB));
```

Comment:
Have I missed something why `Unique Diff BASE with first` would not work?
```suggestion
                        tuples.Add((revBase,
                            _diffBaseToB.Text + GetDescriptionForRevision(selectedRev.ObjectId), uniqueBaseToB));
                        tuples.Add((revBase,
                            _diffBaseToB.Text + GetDescriptionForRevision(firstRev.ObjectId), uniqueBaseToA));
```

Reply:
Unique Diff BASE with first requires that the datastructure changes and affects all forms. (Well the list could be displayed, but no actions would work, as selected/Revision is used for second revisions in diffs it would be confusing.)

As other commits need to be merged prior to #7899, I hope this is a way to speedup reviews...